src/utils/edit/selectionRange.ts

```diff
@@ -107,7 +107,13 @@ export function setSelectionRange(
 
   const selection = element.ownerDocument.getSelection()
   // istanbul ignore else
-  if (selection) {
+  if (
+    selection &&
+    // Skip setting the range for <input> and <textarea> because
+    // the parent <div> will be selected.
+    !isElementType(element, 'input') &&
+    !isElementType(element, 'textarea')
```

Comment:
Could you move the check to above line 99 and just `return` there.
I'd prefer to have the the check and comment stick out.

> the parent `<div>` will be selected.

The selecting of the parent I mentioned on the issue happened on the `click`.
I think the selection is not changed at all. Might be that the selection can be set differently.

Unless you discovered that the typing does move the Range, let's not confuse future devs.
Something like:
> Moving the selection inside `<input>` or `<textarea>` does not alter the document Selection.

Reply:
Updated in 9efb5b1